src/d3a/models/state.py

```diff
@@ -44,14 +44,55 @@
 
 class PVState:
     def __init__(self):
-        self.available_energy_kWh = {}
+        self._available_energy_kWh = {}
+        self._energy_production_forecast_kWh = {}
+
+    def get_energy_production_forecast_kWh(self, time_slot, default_value=None):
+        assert self._energy_production_forecast_kWh[time_slot] >= -FLOATING_POINT_TOLERANCE
+        if default_value is not None:
+            return self._energy_production_forecast_kWh.get(time_slot, default_value)
+        return self._energy_production_forecast_kWh[time_slot]
+
+    def get_available_energy_kWh(self, time_slot, default_value=None):
+        assert self._available_energy_kWh[time_slot] >= -FLOATING_POINT_TOLERANCE
+        if default_value is not None:
+            return self._available_energy_kWh.get(time_slot, default_value)
```

Comment:
Here `.get` is used to access value from `_available_energy_kWh ` dictionary for a `time_slot ` and if not found, a  `default_value ` would return BUT two lines above via assert would already fail for cases when the key doesn't exist.

Reply:
Agreed! I'll disable the assert in case the caller expects a default_value. Thanks for catching this!